### PR TITLE
History Table v2

### PR DESCRIPTION
**Pull request by vishwajeetraj11**
Table layout update for pondToMPond and MPondToPond

### CODE REVIEW

File: src/routes/v2/bridge/+layout.svelte [hunk 1]
```diff
@@ -72,7 +72,9 @@
 
 {#if $chainStore.isValidChain && chainSupported}
 	<PageWrapper>
-		<slot />
+		<div class="mx-[28px] flex flex-1 flex-col">
```
Comment (souvikmishra) [inline src/routes/v2/bridge/+layout.svelte:75]: is this needed here for the bridge as a whole? or is it specific to history pages?

File: src/routes/v2/bridge/pondToMPondHistory/+page.svelte [hunk 2]
```diff
@@ -38,3 +41,8 @@
 </script>
 
 <PondToMPondHistoryData bind:loading bind:historyData />
+<!-- historyData={[
+	...pondToMPondHistoryData,
```
Comment (souvikmishra) [inline src/routes/v2/bridge/pondToMPondHistory/+page.svelte:45]: maybe remove the commented code

File: src/routes/v2/bridge/mPondToPondHistory/+page.svelte [hunk 3]
```diff
@@ -35,6 +37,581 @@
 		previousChainId = null;
 		previousWalletAddress = '';
 	}
+	const pondToMPondHistoryData = [
```
Comment (souvikmishra) [inline src/routes/v2/bridge/mPondToPondHistory/+page.svelte:40]: ig this is some mock data to fill in the table? maybe we can remove this?

File: src/routes/v2/bridge/+layout.svelte (hunk 1 above)
Comment (vishwajeetraj11) [reply src/routes/v2/bridge/+layout.svelte:75]: history pages.

Comment (souvikmishra) [reply src/routes/v2/bridge/+layout.svelte:75]: then we can have it for just the history pages ig

File: src/routes/v2/bridge/pondToMPondHistory/+page.svelte (hunk 2 above)
Comment (vishwajeetraj11) [reply src/routes/v2/bridge/pondToMPondHistory/+page.svelte:45]: removed

File: src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte [hunk 4]
```diff
@@ -1,60 +1,74 @@
...
+				>
+				<td class={tableCellClasses.rowNormal}
 					>{bigNumberToString(row.pondConverted, DEFAULT_CURRENCY_DECIMALS, POND_PRECISIONS)}</td
 				>
-				<td class={tableCellClasses.row}
+				<td class={tableCellClasses.rowNormal}
 					>{bigNumberToString(row.mpondReceived, DEFAULT_CURRENCY_DECIMALS, MPOND_PRECISIONS)}</td
 				>
-				<td class={tableCellClasses.row}>
+				<!-- <td class={tableCellClasses.rowNormal}>
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte:64]: removed.

File: src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte [hunk 5]
```diff
@@ -1,60 +1,74 @@
 <script lang="ts">
-	import { tableCellClasses } from '$lib/atoms/componentClasses';
+	import { tableCellClasses } from '$lib/atoms/v2/componentClasses';
 	import TxnHashText from '$lib/components/texts/TxnHashText.svelte';
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte:3]: PondToMPondHistoryData.svelte is already duplicated 
inside v2 folder in page-components

File: src/lib/page-components/v2/bridge/history/MPondToPondHistoryData.svelte [hunk 6]
```diff
@@ -1,36 +1,52 @@
 <script lang="ts">
 	import type { MPondToPondHistoryDataModel } from '$lib/types/bridgeComponentType';
-	import { MPOND_TO_POND_TABLE_HEADER } from '$lib/utils/constants/bridgeConstants';
-	import HistoryTableCommon from '$lib/page-components/bridge/history/HistoryTableCommon.svelte';
-	import MPondTableRow from '$lib/page-components/bridge/history/MPondTableRow.svelte';
+	import { MPOND_TO_POND_TABLE_HEADER } from '$lib/utils/constants/v2/bridgeConstants';
+	import HistoryTableCommon from '$lib/page-components/v2/bridge/history/HistoryTableCommon.svelte';
+	// import MPondTableRow from '$lib/page-components/bridge/history/MPondTableRow.svelte';
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/history/MPondToPondHistoryData.svelte:5]: resolved.

File: src/lib/atoms/table/Table.svelte [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-	import TableHeadingText from '$lib/components/texts/TableHeadingText.svelte';
+	import TableHeadingText from '$lib/components/v2/texts/TableHeadingText.svelte';
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/atoms/table/Table.svelte:2]: resolved

File: src/lib/components/texts/TxnHashText.svelte [hunk 8]
```diff
@@ -8,6 +8,6 @@
 </script>
 
 <div class="flex items-center justify-center gap-2" data-testid="txn-hash-text">
-	{shortenText(txnHash, startInt, endInt)}
+	<!-- {shortenText(txnHash, startInt, endInt)} -->
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/texts/TxnHashText.svelte:11]: resolved.
no need to update

File: src/lib/atoms/v2/table/Table.svelte [hunk 9]
```diff
@@ -17,7 +17,7 @@
 				{#each tableHeading as columnHeading, i}
 					<th class="py-[26px]">
 						<TableHeadingText
-							styleClass="{headingStyleClass} {i === 0 ? 'ml-3' : ''}"
+							styleClass="{headingStyleClass} {i === 0 ? 'ml-4' : ''}"
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/atoms/v2/table/Table.svelte:20]: done

File: src/lib/components/v2/table-cells/TableDataCell.svelte [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<script lang="ts">
+	import { tableCellClasses } from '$lib/atoms/componentClasses';
+	export let styleClass = '';
+</script>
+
+<td class="{styleClass} {tableCellClasses.rowNormal}" data-testid="table-data-cell">
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/TableDataCell.svelte:6]: done

File: src/lib/components/v2/table-cells/TableDataWithButton.svelte [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+	import { tableCellClasses } from '$lib/atoms/componentClasses';
+	export let styleClass = '';
+	export let firstRow: boolean = false;
+</script>
+
+<td class="{styleClass} {firstRow ? 'pl-4' : ''} h-[64px]" data-testid="table-data-with-button">
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/TableDataWithButton.svelte:7]: done

File: src/lib/components/v2/table-cells/__snapshots__/TableCells.dom.spec.ts.snap [hunk 12]
```diff
@@ -0,0 +1,37 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`Table Cells > renders TableDataCell successfully 1`] = `
+<body>
+  <div>
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/__snapshots__/TableCells.dom.spec.ts.snap:5]: okay

File: src/lib/page-components/v2/bridge/buttons/MPondConversionCycleButton.svelte [hunk 13]
```diff
@@ -12,6 +11,11 @@
 </script>
 
 <ModalButton size="tiniest" variant="text" {modalFor}>
-	<HistoryDataIconButton src={staticImages.Cycle} text="See cycle" />
+	<!-- <HistoryDataIconButton src={staticImages.Cycle} text="See cycle" /> -->
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/buttons/MPondConversionCycleButton.svelte:14]: removed

File: src/lib/page-components/v2/bridge/buttons/MPondConversionCycleButton.svelte [hunk 14]
```diff
@@ -12,6 +11,11 @@
...
+	<div
+		class="flex h-[45px] w-[45px] items-center justify-center rounded-full border border-[#D9DADE] hover:bg-[#F0F0F0]"
+	>
+		<img src={staticImages.conversionCycleIcon} alt="" />
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/buttons/MPondConversionCycleButton.svelte:18]: added

File: src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte [hunk 15]
```diff
@@ -2,29 +2,24 @@
...
 	import { staticImages } from '$lib/components/images/staticImages';
 	import type { MPondToPondHistoryDataModel } from '$lib/types/bridgeComponentType';
-	import MPondConversionHistoryModal from '$lib/page-components/bridge/modals/MPondConversionHistoryModal.svelte';
-	import HistoryDataIconButton from '$lib/page-components/bridge/sub-components/HistoryDataIconButton.svelte';
+	import MPondConversionHistoryModal from '$lib/page-components/v2/bridge/modals/MPondConversionHistoryModal.svelte';
 
 	export let conversionHistory: MPondToPondHistoryDataModel['conversionHistory'];
 	export let modalFor: string;
+
+	const classes =
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte:10]: that would be repeating the the same classes twice, i 've wrapped the below usecases in cn

File: src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte [hunk 16]
```diff
@@ -2,29 +2,24 @@
...
 {#if !conversionHistory?.length}
-	<HistoryDataIconButton
-		variant="disabled"
-		src={staticImages.History}
-		imgWidth={14}
-		text="History"
-		disabled={true}
-	/>
+	<div class="{classes} cursor-not-allowed opacity-50">
+		<img src={staticImages.historyIcon} alt="" />
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte:17]: done

File: src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte [hunk 17]
```diff
@@ -2,29 +2,24 @@
...
+	</div>
 {:else}
 	<ModalButton variant="text" size="tiniest" {modalFor}>
-		<HistoryDataIconButton
-			text="History"
-			variant="primary"
-			src={staticImages.History}
-			imgWidth={14}
-		/>
+		<div class="{classes} hover:bg-[#F0F0F0]">
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/buttons/MPondConversionHistoryButton.svelte:21]: done

File: src/lib/page-components/v2/bridge/history/HistoryTableCommon.svelte [hunk 18]
```diff
@@ -23,14 +23,25 @@
...
+		class="mb-8 flex h-[56px] w-[56px] items-center justify-center rounded-full border border-[#D9DADE] bg-white"
+		href={tableTitle.backButton.href}
+	>
+		<img src={staticImages.backIcon} alt="Back Icon" />
+	</a>
+	<PageTitle title={tableTitle.title} />
+</div>
 <div
-	class="card rounded-lg bg-base-100 {fullWidth ? 'max-w-full' : 'sm:max-w-full md:max-w-[66.66%]'}"
+	class="card rounded-[18px] bg-base-100 {fullWidth
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/history/HistoryTableCommon.svelte:42]: done

File: src/lib/components/texts/TxnHashText.svelte [hunk 19]
```diff
@@ -8,6 +8,5 @@
 </script>
 
 <div class="flex items-center justify-center gap-2" data-testid="txn-hash-text">
-	{shortenText(txnHash, startInt, endInt)}
 	<TxnIcon {txnHashUrl} />
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/texts/TxnHashText.svelte:11]: fixed. didnt need this component.

File: src/lib/components/v2/table-cells/TableDataCell.svelte [hunk 20]
```diff
@@ -0,0 +1,9 @@
+<script lang="ts">
+	import { tableCellClasses } from '$lib/atoms/componentClasses';
+	import { cn } from '$lib/utils/helpers/commonHelper';
+	export let styleClass = '';
+</script>
+
+<td class={cn(styleClass, tableCellClasses.rowNormal)} data-testid="table-data-cell">
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/TableDataCell.svelte:7]: its an existing implementation, i've just moved this file to v2 folder.
should this be fixed in this PR?

[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/TableDataCell.svelte:7]: image

File: src/lib/components/v2/table-cells/TableDataWithButton.svelte [hunk 21]
```diff
@@ -0,0 +1,17 @@
+<script lang="ts">
+	import { tableCellClasses } from '$lib/atoms/componentClasses';
+	import { cn } from '$lib/utils/helpers/commonHelper';
+	export let styleClass = '';
+	export let firstRow: boolean = false;
+</script>
+
+<td class={cn(styleClass, firstRow ? 'pl-4' : '', 'h-[64px]')} data-testid="table-data-with-button">
```
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/components/v2/table-cells/TableDataWithButton.svelte:8]: removed it.

File: src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (vishwajeetraj11) [reply src/lib/page-components/v2/bridge/history/PondToMPondHistoryData.svelte:3]: that component was not needed here it was an unused import, removed it.

File: src/routes/v2/bridge/+layout.svelte (hunk 1 above)
Comment (vishwajeetraj11) [reply src/routes/v2/bridge/+layout.svelte:75]: yes

**Review by souvikmishra — APPROVED** [review]

**Pull request merged**